californium-core/src/main/java/org/eclipse/californium/core/coap/CoAP.java

```diff
@@ -135,6 +135,30 @@ public static boolean isSupportedScheme(final String uriScheme) {
 				CoAP.COAP_SECURE_URI_SCHEME.equalsIgnoreCase(uriScheme) ||
 				CoAP.COAP_SECURE_TCP_URI_SCHEME.equalsIgnoreCase(uriScheme);
 	}
+	
+	/**
+	 * Get default port for provided uri scheme.
+	 * 
+	 * @param uriScheme uri scheme for default port
+	 * @return default port, or 0, if no default port is defined for the provided uri scheme.
```

Comment:
IMHO we should throw an `IllegalArgumentException` in case of an unsupported scheme (instead of returning 0 which needs to be checked by the client afterwards). Client code can use `isSupportedScheme()` before invoking this method then.